### PR TITLE
Make neutron-l3 deployable on SLE12

### PR DESCRIPTION
**Pull request by aplanas**

### CODE REVIEW

File: chef/cookbooks/neutron/recipes/l3.rb [hunk 1]
```diff
@@ -53,16 +53,31 @@
...
 
+# Enable ip forwarding on network node for SLE12
+ruby_block "edit /etc/sysctl.d/99-sysctl.conf for net.ipv4.ip_forward" do
+  block do
+    rc = Chef::Util::FileEdit.new("/etc/sysctl.d/99-sysctl.conf")
+    rc.search_file_replace_line(/^net.ipv4.ip_forward =/, 'net.ipv4.ip_forward = 1')
+    rc.write_file
+  end
+  only_if { node[:platform] == "suse" && node[:platform_version].to_f >= 12.0 }
+end
```
Comment (vuntz) [inline chef/cookbooks/neutron/recipes/l3.rb]: I guess you need to put this code after `directory "create /etc/sysctl.d for enable-ip_forward" do`, or is the directory just always existing?

Comment (aplanas) [inline chef/cookbooks/neutron/recipes/l3.rb]: AFAIK always exists for SLE[SD]12

**Comment by jsuchome** [comment]
+1

**Comment by rhafer** [comment]
@aplanas In genereal this looks good to me. But I'd like to know more about the testing you did on this. Which networking drivers have you tested this with?
Can you probably test this with ml2 + openvswitch + vlan? I am asking because there is some code in the ```common_agent``` recipe that creates a new init script which reassigns IP addresses from physical interfaces to OVS bridges. Can you verify that this really works (and survives a reboot of the l3 node)?

**Comment by aplanas** [comment]
@rhafer Tested with ml2 pluging and linuxbridge (http://crowbar.c16.cloud.suse.de:3000/crowbar/neutron/1.0/proposals/default)

**Comment by aplanas** [comment]
Tested witn ovs + gre

**Comment by aplanas** [comment]
Tested with ovs + vlan in an unpatched cloud instance.

**Comment by mjura** [comment]
:+1: looks good to me

**Pull request merged**